File: VirtualLab_MatLab/SimPla_MATLAB/tokamaks/equilibrium/Tokalab_Scenario.m
function config = Tokalab_Scenario(separatrix,toroidal_current_method)
    
    %% Here we define the values for the target separatrix

    if separatrix == 1
        % single-null standard tokalab scenario
        config.separatrix.scenario = 1;
        config.separatrix.method = 1;
        config.separatrix.k1 = 1.7;
        config.separatrix.k2 = 2;
        config.separatrix.d1 = 0.5;
        config.separatrix.d2 = 0.5;
        config.separatrix.gamma_n_1 = 0;
        config.separatrix.gamma_n_2 = pi/3;
        config.separatrix.gamma_p_1 = 0;
        config.separatrix.gamma_p_2 = pi/6;

        config.separatrix.a = 2;
        config.separatrix.R0 = 6;
        config.separatrix.Z0 = 0;

    elseif separatrix == 2
        % Double-null standard tokalab scenario
        config.separatrix.scenario = 2;
        config.separatrix.method = 1;
        config.separatrix.k1 = 1.7;
        config.separatrix.k2 = 1.7;
        config.separatrix.d1 = 0.5;
        config.separatrix.d2 = 0.5;
        config.separatrix.gamma_n_1 = pi/3;
        config.separatrix.gamma_n_2 = pi/3;
        config.separatrix.gamma_p_1 = pi/6;
        config.separatrix.gamma_p_2 = pi/6;
         
        config.separatrix.a = 2;
        config.separatrix.R0 = 6;
        config.separatrix.Z0 = 0;

      elseif separatrix == 3
        % Negative Triangularity
        config.separatrix.scenario = 3;
        config.separatrix.method = 1;
        config.separatrix.k1 = 1.7;
        config.separatrix.k2 = 2;
        config.separatrix.d1 = -0.5;
        config.separatrix.d2 = -0.5;
        config.separatrix.gamma_n_1 = 0;
        config.separatrix.gamma_n_2 = pi/6;
        config.separatrix.gamma_p_1 = 0;
        config.separatrix.gamma_p_2 = pi/3;

        config.separatrix.a = 1.9;
        config.separatrix.R0 = 5.9;
        config.separatrix.Z0 = 0.1;

    end

    %% Define the toroidal current and field parameters

    if toroidal_current_method == 1
        % this method implement the function in (reference)
        config.toroidal_current.method = 1;
        
        config.toroidal_current.Bt = 3;
        config.toroidal_current.Ip = -12e6;

        config.toroidal_current.alpha_1 = 2;
        config.toroidal_current.alpha_2 = 2;
        config.toroidal_current.beta_0 = 0.5;
    
        config.toroidal_current.lambda = 1;
        
    elseif toroidal_current_method == 2
        % New method to be implemented
    end



end

File: VirtualLab_Python/SimPla_Python/tokamaks/equilibrium/Tokalab_Scenario.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jun  2 15:00:14 2025

@author: ricca
"""

import math

class Separatrix:
    def __init__(self, separatrix=1):
        if separatrix == 1:
            # single null
            self.scenario = 1
            self.method = 1
            self.k1 = 1.7
            self.k2 = 2
            self.d1 = 0.5
            self.d2 = 0.5
            self.gamma_n_1 = 0
            self.gamma_n_2 = math.pi / 3
            self.gamma_p_1 = 0
            self.gamma_p_2 = math.pi / 6
            
            self.a = 2
            self.R0 = 6
            self.Z0 = 0
            
        elif separatrix == 2:
            # double null
            self.scenario = 1
            self.method = 1
            self.k1 = 1.7
            self.k2 = 1.7
            self.d1 = 0.5
            self.d2 = 0.5
            self.gamma_n_1 = math.pi / 3
            self.gamma_n_2 = math.pi / 3
            self.gamma_p_1 = math.pi / 6
            self.gamma_p_2 = math.pi / 6
            
            self.a = 2
            self.R0 = 6
            self.Z0 = 0
            
        elif separatrix == 3:
            # double null
            self.scenario = 3
            self.method = 1
            self.k1 = 1.7
            self.k2 = 2
            self.d1 = -0.5
            self.d2 = -0.5
            self.gamma_n_1 = 0
            self.gamma_n_2 = math.pi / 6
            self.gamma_p_1 = 0
            self.gamma_p_2 = math.pi / 3
            
            self.a = 2
            self.R0 = 6
            self.Z0 = 0

class Toroidal_Current:
    def __init__(self, method=1):
        if method == 1:
            self.method = 1
            self.Bt = 5
            self.Ip = -12e6
            self.alpha_1 = 2
            self.alpha_2 = 2
            self.beta_0 = 0.5
            self.lambda_ = 1
        elif method == 2:
            # Placeholder for a new method
            pass


class Tokalab_Scenario:
    def __init__(self, separatrix=1, toroidal_current_method=1):
        self.separatrix = Separatrix(separatrix)
        self.toroidal_current = Toroidal_Current(toroidal_current_method)